### ISSUE TITLE
Merge code for text rendering in visuals and collections

### ISSUE THREAD

**Issue by almarklein** [vispy/vispy#798]

**Comment by Eric89GXL** [comment]
@rougier I looked at the text visual a little bit, and it turns out there is already a font manager in there. So hopefully the merging won't be too difficult.

**Comment by rougier** [comment]
Yes, but I did not know where to put the font manager code in the vispy hierarchy.
Also note that we need a (unique) Agg and SDF font manager.

**Comment by Eric89GXL** [comment]
Ahh, right. We should probably refactor that part to go into `visuals/fonts` or so. The manager can be global, but the SDF renderer cannot. If someone has multiple contexts, or creates and destroys contexts, the SDF renderer will need to switch/change appropriately.

**Comment by almarklein** [comment]
I think we should store the renderer on the `canvas.context.shared` object.

**Comment by djhoese** [comment]
I'm removing the milestone for this issue and #799. There is no description on these and I'm not completely sure what they are meant to cover.

**Comment by kmuehlbauer** [comment]
@almarklein Is this issue and #799 still relevant? If so, it would be good to get a bit more detail on both issues.

**Comment by djhoese** [comment]
I think they if I had to guess as I don't think anyone has touched the collections since they were copied from glumpy. Both of these issues are meant to consolidate the functionality between the glumpy-style collections and vispy-style visuals.

**Comment by almarklein** [comment]
I have no idea what the state of collections is in vispy. If we decided to drop that effort, we can close these issues.

**Comment by djhoese** [comment]
Collections as a concept of creating a single OpenGL Program with a lot of separate elements (a ton of spheres) that performs well and doesn't make it difficult for users is still desired (in my opinion). Every time I look at the existing collections stuff from glumpy I get confused and walk away. At some point they need to be ported to work like a Visual (probably) so maybe this should stay open.

**Comment by rougier** [comment]
Collection is mostly a trick to encapsulate object parameters inside a texture even though we don't know what kind of parameters the user want. I only found this way with GL ES 2.0. With modern GL, it would be much easier.
